crates/matrix-sdk/src/room/request_to_join.rs

```diff
@@ -0,0 +1,95 @@
+use std::sync::Arc;
+
+use ruma::{
+    events::room::member::OriginalSyncRoomMemberEvent, EventId, OwnedEventId, OwnedMxcUri,
+    OwnedUserId, RoomId,
+};
+
+use crate::{room::RoomMember, Error, Room};
+
+/// A request to join a room with `knock` join rule.
+#[derive(Debug, Clone)]
+pub struct RequestToJoinRoom {
+    room: Arc<Room>,
```

Comment:
(This pattern of using an `Arc<Room>` lets me think that you've ran into the `Room` identity issue I've spotted above. We usually don't store `Room` in `Arc`, because we shouldn't have to.)

Reply:
Nice, I think I got over-paranoid with not cloning stuff, so I'm reverting this to using a `Room` directly and cloning it as needed.